### PR TITLE
Cleaner configuration management

### PR DESCRIPTION
**Pull request by joedevivo**
- put explicit defaults in app.src
- don't recompute Server header per request
- Pull webmachine version for Server header from application metadata

Despite the branch name, it turns out that I just did some cleanup that could lead to lager3 in the future.

### CODE REVIEW

File: src/webmachine_mochiweb.erl [hunk 1]
```diff
@@ -121,20 +121,21 @@ new_webmachine_req(Request) ->
...
-                       Method,
-                       Scheme,
-                       Version,
-                       mochiweb_request:get(headers, Request),
-                       mochiweb_request:get(raw_path, Request));
-        undefined ->
-            {
+    {Headers, RawPath} =
+        case application:get_env(webmachine, rewrite_module) of
+            {ok, undefined} ->
```
Comment (seancribbs) [inline src/webmachine_mochiweb.erl:126]: Is this `{ok, undefined}` or just `undefined`?

File: src/webmachine_app.erl [hunk 2]
```diff
@@ -48,3 +50,27 @@ start(_Type, _StartArgs) ->
...
+    ok.
+
+
+set_default_server_name() ->
+    {webmachine, _, Version} =
+        lists:keyfind(webmachine, 1, application:loaded_applications()),
+    ServerName =
+        lists:flatten(
+          io_lib:format(
+            "MochiWeb/1.1 WebMachine/~s (~s)",
```
Comment (seancribbs) [inline src/webmachine_app.erl]: We should probably produce the correct mochiweb version while we're at it.

File: src/webmachine_mochiweb.erl (hunk 1 above)
Comment (seancribbs) [reply src/webmachine_mochiweb.erl:126]: Oh, I see that you're setting it to undefined in the app.src.

**Review by seancribbs — CHANGES_REQUESTED** [review]
Minor nitpicks, yadda yadda

File: src/webmachine_app.erl (hunk 2 above)
Comment (joedevivo) [reply src/webmachine_app.erl]: Pushed, although hopefully we'll need to dynamically determine whether to include MochiWeb at all here :D

File: src/webmachine_mochiweb.erl (hunk 1 above)
Comment (joedevivo) [reply src/webmachine_mochiweb.erl:126]: No implied configuration! :D

**Review by seancribbs — APPROVED** [review]
MOCHIWEB'D!!!!

**Pull request merged**